docs/python/how-to-guides/sysadmin/setuptools-deployment.md

```diff
@@ -0,0 +1,407 @@
+
+First, install the build tool:
+
+```shell
+pip install build
+```
+
+Then build your package:
+
+```shell
+cd my_dh_tool
+python -m build
+```
+
+This creates two files in the `dist/` directory:
+
+- A source distribution (`.tar.gz`)
+- A wheel distribution (`.whl`)
+
+The wheel file can be distributed and installed on other systems.
+
+## Install the package
+
+### For development
+
+Install your package in editable mode for development:
+
+```shell
+pip install -e .
+```
+
+This allows you to make changes to your code without reinstalling.
+
+### For regular use
+
+Install from the built wheel:
+
+```shell
+pip install dist/my_dh_tool-0.1.0-py3-none-any.whl
+```
+
+### Using pipx (recommended for command-line tools)
+
+[pipx](https://pipx.pypa.io/) is the recommended way to install Python command-line tools, as it installs them in isolated environments:
+
+```shell
+pipx install .
+```
+
+Or from the wheel:
+
+```shell
+pipx install dist/my_dh_tool-0.1.0-py3-none-any.whl
+```
+
+After installation, the `my-dh-tool` command is available system-wide (or within your virtual environment).
+
+### Run without installing
+
+You can also run your tool directly from the source directory:
+
+```shell
+python -m my_dh_tool input_data.csv
+```
+
+Or use `pipx run` for temporary execution:
+
+```shell
+pipx run --spec . my-dh-tool input_data.csv
+```
+
+## Run the command-line tool
+
+Once installed, you can run your tool from any directory:
+
+```shell
+my-dh-tool input_data.csv
+```
+
+The command executes the `app()` function in your `cli.py` file.
+
+## Advanced configuration
+
+### Multiple entry points
+
+You can define multiple command-line scripts in one package:
+
+```toml
+[project.scripts]
+my-dh-tool = "my_dh_tool.cli:app"
+my-dh-process = "my_dh_tool.processor:process"
+my-dh-report = "my_dh_tool.reporter:generate_report"
+```
+
+### Advanced argument parsing
+
+For more sophisticated CLIs, consider using [typer](https://typer.tiangolo.com/) or [click](https://click.palletsprojects.com/):
+
+```python syntax
+import typer
+from typing_extensions import Annotated
+
+
+def app(
+    input_file: Annotated[str, typer.Argument(help="Input file to process")],
+    output: Annotated[str, typer.Option(help="Output file")] = None,
+    verbose: Annotated[bool, typer.Option(help="Verbose output")] = False,
+):
+    """Process data with Deephaven."""
+    if verbose:
+        typer.echo(f"Processing {input_file}...")
+
+    # Process data using Deephaven
+    result = process_data(input_file)
+
+    if output:
+        typer.echo(f"Results saved to {output}")
+
+
+if __name__ == "__main__":
+    typer.run(app)
+```
+
+Add `typer` to your dependencies:
+
+```toml
+dependencies = [
+  "deephaven-server>=0.35.0",
+  "typer>=0.9.0",
+]
+```
+
+### Configuration files
+
+For more complex tools, consider using configuration files:
+
+```python syntax
+import configparser
+import os
+
+
+def load_config():
```

Comment:
We should make an effort to typehint all funcs in this document.  It is our project standard.  (This one may be harder)